docs/src/reference/variables.md

```diff
@@ -0,0 +1,119 @@
+# Variables
+
+MACH composer support the usage of variables in a configuration file.
+
+The following types are supported;
+
+- [`${component.}`](#component) component output references
+- [`${var.}`](#var) variables file values
+- [`${env.}`](#env) environment variables value
+- [`${include.}`](#include) file includes
+
+## Example
+
+```yaml
+mach_composer:
+  version: 1.0.0
+global:
+  environment: ${env.MACH_ENVIRONMENT}
+  cloud: aws
+sites:
+  - identifier: my-site
+    aws:
+      account_id: 1234567890
+      region: eu-central-1
+    endpoints:
+      public: api.tst.mach-example.net
+    components:
+      - name: infra
+      - name: payment
+        variables:
+          sns_topic: ${components.infra.sns_topic_arn}
+        secrets:
+          stripe_secret_key: ${var.stripe_secret}
+components: ${include(components.yml)}
+```
+
+- `${components.infra.sns_topic_arn}` uses the `sns_topic_arn` Terraform output as a value for the payment component
+- `${var.stripe_secret}` reads the `stripe_secret` from a variables file
+- `${include(components.yml)}` includes `components.yml` and injects it in the configuration
+
+## `component`
+**Usage** `${component.<component-name>.<output-value>}`
+
+You can use this to refer to any [Terraform output](https://www.terraform.io/docs/language/values/outputs.html) that another component has defined.
+
+So for example if a component called "infra" has the following outputs:
+
+```terraform
+# outputs.tf
+
+output "sns_topic_arn" {
+    value = aws_sns_topic.main.arn
+}
+
+output "psql_server" {
+    value = {
+        endpoint = aws_db_instance.main.endpoint
+        arn = aws_db_instance.main.arn
+    }
+}
+```
+
+These can then be used in the configuration:
+
+```yaml
+components:
+  - name: order-notifier
+    variables:
+      sns_topic: ${component.infra.sns_topic_arn}
+      database_endpoint: ${component.psql_server.endpoint}
```

Comment:
should this say: `component.infra.psql_server.endpoint` as it is part of the same `infra` component?

Reply:
And: perhaps this is not the best example, as components should contain their own infrastructure, in order to be standalone :)